src/julia-serial/DataFormats/data_formats.jl

```diff
@@ -0,0 +1,173 @@
+        end
+        resize!(self.data, newsize)
+    end
+
+
+
+    """
+    FedRawDataCollection struct
+
+    Represents a collection of FedRawData objects.
+
+    Fields:
+    - data::Vector{FedRawData}: Vector of FedRawData objects.
+
+    Constructor:
+    - FedRawDataCollection(in::FedRawDataCollection): Copy constructor.
+    """
+
+
+
+    mutable struct FedRawDataCollection
+        data::Vector{FedRawData}  # Vector of FedRawData
+        """
+        Copy constructor.
+        """
+        FedRawDataCollection(in::FedRawDataCollection) = new(copy(in.data)) # copy constructor
+        FedRawDataCollection(data::Vector{FedRawData}) = new(data)
+        FedRawDataCollection() = new(Vector{FedRawData}(undef,MAX_FED_ID))
+    end
+
+    """
+    Swap function for FedRawDataCollection.
+    """
+    function swap(a::FedRawDataCollection, b::FedRawDataCollection)
+        a.data, b.data = b.data, a.data
+    end
+    """ 
+    function for getting FedRawData
+    """
+    function FedData(self::FedRawDataCollection,Fedid :: Integer)
+        return self.data[Fedid]
+    end
+
+    
+    mutable struct SiPixelRawDataError
+        errorWord32::UInt32
+        errorWord64::UInt64
+        errorType::Int
+        FedId::Int
+        errorMessage::String
+        """
+        Constructor for 32-bit error word
+        """
+        
+        function SiPixelRawDataError(errorWord32::UInt32, errorType::Int, FedId::Int)
+            new(errorWord32, UInt64(0), errorType, FedId, "")
+        end
+        """
+        Constructor with 64-bit error word and type included (header or trailer word)
+        """
+        
+        function SiPixelRawDataError(errorWord64::UInt64, errorType::Int, FedId::Int)
+            new(UInt32(0), errorWord64, errorType, FedId, "")
+        end
+    end
+    function setType!(error::SiPixelRawDataError, errorType::Int)
+        error.errorType = errorType
+        setMessage!(error)
+    end
+    function setFedId!(error::SiPixelRawDataError, FedId::Int)
+        error.FedId = FedId
+    end
+    function setMessage!(error::SiPixelRawDataError)
+        error.errorMessage = errorTypeMessage(error.errorType)
+    end
+    function errorTypeMessage(errorType::Int)::String
+        return Dict(
+            25 => "Error: Disabled Fed channel (ROC=25)",
+            26 => "Error: Gap word",
+            27 => "Error: Dummy word",
+            28 => "Error: FIFO nearly full",
+            29 => "Error: Timeout",
+            30 => "Error: Trailer",
+            31 => "Error: Event number mismatch",
+            32 => "Error: Invalid or missing header",
+            33 => "Error: Invalid or missing trailer",
+            34 => "Error: Size mismatch",
+            35 => "Error: Invalid channel",
+            36 => "Error: Invalid ROC number",
+            37 => "Error: Invalid dcol/pixel address",
+        )[errorType, "Error: Unknown error type"]
+    end

```

Comment:
```suggestion
    function errorTypeMessage(errorType::Int)
        err_dict = Dict(
            25 => "Error: Disabled Fed channel (ROC=25)",
            26 => "Error: Gap word",
            27 => "Error: Dummy word",
            28 => "Error: FIFO nearly full",
            29 => "Error: Timeout",
            30 => "Error: Trailer",
            31 => "Error: Event number mismatch",
            32 => "Error: Invalid or missing header",
            33 => "Error: Invalid or missing trailer",
            34 => "Error: Size mismatch",
            35 => "Error: Invalid channel",
            36 => "Error: Invalid ROC number",
            37 => "Error: Invalid dcol/pixel address",
        )
        return get(err_dict, errorType, "Error: Unknown error type")
    end

```

the original code doesn't work
```julia
julia> a = [1=>"213"]
1-element Vector{Pair{Int64, String}}:
 1 => "213"

julia> a[2, "asd"]
ERROR: ArgumentError: invalid index: "asd" of type String
Stacktrace:
```